host/types/types.go

```diff
@@ -68,12 +68,39 @@ type ContainerConfig struct {
 	Cmd         []string          `json:"cmd,omitempty"`
 	Env         map[string]string `json:"env,omitempty"`
 	Mounts      []Mount           `json:"mounts,omitempty"`
+	Volumes     []VolumeBinding   `json:"volumes,omitempty"`
 	Ports       []Port            `json:"ports,omitempty"`
 	WorkingDir  string            `json:"working_dir,omitempty"`
 	Uid         int               `json:"uid,omitempty"`
 	HostNetwork bool              `json:"host_network,omitempty"`
 }
 
+func (x ContainerConfig) Merge(y ContainerConfig) ContainerConfig {
+	x.TTY = x.TTY || y.TTY
+	x.Stdin = x.Stdin || y.Stdin
+	x.Data = x.Data || y.Data
+	if y.Entrypoint != nil {
+		x.Entrypoint = y.Entrypoint
+	}
+	if y.Cmd != nil {
+		x.Cmd = y.Cmd
+	}
+	for k, v := range y.Env {
+		x.Env[k] = v
+	}
+	x.Mounts = append(x.Mounts, y.Mounts...)
+	x.Volumes = append(x.Volumes, y.Volumes...)
+	x.Ports = append(x.Ports, y.Ports...)
```

Comment:
I stand corrected... that example only works because the append call happens to be allocating a new copy.  Here's an example with capacity preallocated which behaves differently: http://play.golang.org/p/5KyiyTKSTK